src/sentry/search/eap/spans.py

```diff
@@ -310,6 +310,61 @@ def _resolve_having(
             return parsed_terms[0], resolved_contexts
         return None, []
 
+    def _handle_virtual_column_terms(
+        self,
+        term: event_search.SearchFilter,
+        resolved_column: ResolvedColumn,
+        context: VirtualColumnContext | None,
+    ) -> tuple[ResolvedColumn, float | datetime | Sequence[float] | Sequence[str]]:
+        raw_value = term.value.raw_value
+        if context is None or isinstance(raw_value, datetime):
+            return resolved_column, raw_value
+        raw_value = cast(str | int | list[str], raw_value)
+
+        # Convert the term to the expected values
+        final_raw_value: str | int | list[str] | list[int] = []
+        reversed_context = {v: k for k, v in context.value_map.items()}
+        if term.value.is_wildcard():
+            # Avoiding this for now, but we could theoretically do a wildcard search on the resolved contexts
+            raise InvalidSearchQuery(f"Cannot use wildcards with {term.key}")
+        elif isinstance(raw_value, list):
+            new_value = []
+            for raw_iterable in raw_value:
+                if term.key.name not in constants.PROJECT_FIELDS and raw_iterable == "Unknown":
+                    # Avoiding this for now, while this could work with the Unknown:"" mapping
+                    # But that won't work once we use the VirtualColumnContext.default_value
+                    raise InvalidSearchQuery(
+                        "Using Unknown in an IN filter is not currently supported"
+                    )
+                elif raw_iterable not in reversed_context:
+                    raise InvalidSearchQuery(
+                        constants.REVERSE_CONTEXT_ERROR.format(
+                            raw_value, term.key.name, ", ".join(reversed_context.keys())
+                        )
+                    )
+                else:
+                    new_value.append(reversed_context[raw_iterable])
+            final_raw_value = new_value
+        elif raw_value in reversed_context:
+            final_raw_value = reversed_context[raw_value]
+        else:
+            raise InvalidSearchQuery(
+                constants.REVERSE_CONTEXT_ERROR.format(
+                    raw_value, term.key.name, ", ".join(reversed_context.keys())
+                )
+            )
+
+        # Convert the resolved_column to the underlying column
+        if term.key.name in constants.PROJECT_FIELDS:
+            resolved_column, _ = self.resolve_attribute("project.id")
+            if isinstance(final_raw_value, list):
+                final_raw_value = [int(val) for val in final_raw_value]
+            else:
+                final_raw_value = int(final_raw_value)
+        elif term.key.name == "device.class":
+            resolved_column, _ = self.resolve_attribute("sentry.device.class")
```

Comment:
This will conflict with #83714 where you're trying to make the search resolver applicable to different datasets.

Reply:
😬 ugh right... I'll move this logic to the VCC definitions instead then and merge 83714 first